commit 2142b7f0c6bbe1f9515ce3383de9f7a32a5a025b
Merge: fd2d7a4a3545 afcf5441b9ff
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Mon Mar 21 19:32:04 2022 -0700

    Merge tag 'hardening-v5.18-rc1' of git://git.kernel.org/pub/scm/linux/kernel/git/kees/linux
    
    Pull kernel hardening updates from Kees Cook:
    
     - Add arm64 Shadow Call Stack support for GCC 12 (Dan Li)
    
     - Avoid memset with stack offset randomization under Clang (Marco
       Elver)
    
     - Clean up stackleak plugin to play nice with .noinstr (Kees Cook)
    
     - Check stack depth for greater usercopy hardening coverage (Kees Cook)
    
    * tag 'hardening-v5.18-rc1' of git://git.kernel.org/pub/scm/linux/kernel/git/kees/linux:
      arm64: Add gcc Shadow Call Stack support
      m68k: Implement "current_stack_pointer"
      xtensa: Implement "current_stack_pointer"
      usercopy: Check valid lifetime via stack depth
      stack: Constrain and fix stack offset randomization with Clang builds
      stack: Introduce CONFIG_RANDOMIZE_KSTACK_OFFSET
      gcc-plugins/stackleak: Ignore .noinstr.text and .entry.text
      gcc-plugins/stackleak: Exactly match strings instead of prefixes
      gcc-plugins/stackleak: Provide verbose mode

diff --cc arch/arm64/Kconfig
index a555f409ba95,a6733b972212..8c54de0cf3d6
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@@ -1257,7 -1236,10 +1258,7 @@@ config HW_PERF_EVENT
  	def_bool y
  	depends on ARM_PMU
  
- # Supported by clang >= 7.0
 -config ARCH_HAS_FILTER_PGPROT
 -	def_bool y
 -
+ # Supported by clang >= 7.0 or GCC >= 12.0.0
  config CC_HAVE_SHADOW_CALL_STACK
  	def_bool $(cc-option, -fsanitize=shadow-call-stack -ffixed-x18)